OpenRA.Game/Player.cs

```diff
@@ -62,6 +51,11 @@ struct StanceColors
 		public readonly Shroud Shroud;
 		public readonly FrozenActorLayer FrozenActorLayer;
 
+		readonly Color color;
+
+		/// <returns> Player color with relationship colors applied </returns>
+		public Color Color { get; private set; }
+
```

Comment:
Why have a field `color`, a property `Color`  and a method `GetColor` to access `color`?

Reply:
For a multitude of reasons.

`color` is the actual color assigned to the player, and we do not want any code to use this value. Baring a few exception where a player instance is saved / transferred. For those cases `GetColor` exists

As for `Color`, it is basically a performance cheap way to apply various colors to the player. Currently it only happens once at the start of the game, but it should be dynamic in the future

Reply:
Then this should be `{ get; init; }`

Reply:
Because of the readonly field, but we don't have support for this yet anyway.

Reply:
it isn't readonly, it's set way after initialisation. And in the future we will want to set it multiple times

Reply:
The field above this property.